src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -512,6 +491,98 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+            bool invokedAsExtensionMethod)
+        {
+            Debug.Assert(binder != null);
+
+            // Either the methodOrIndexer is a property, in which case the method used
+            // for optional parameters is an accessor of that property (or an overridden
+            // property), or the methodOrIndexer is used for optional parameters directly.
+            Debug.Assert(((methodOrIndexer.Kind == SymbolKind.Property) && optionalParametersMethod.IsAccessor()) ||
+                (object)methodOrIndexer == optionalParametersMethod);
+
+            // We need to do a fancy rewrite under the following circumstances:
+            // (1) a params array is being used; we need to generate the array. 
+            // (2) there were optional parameters that had no corresponding arguments.
+            //
+            // If neither of those are the case then we can just take an early out.
+
+            ImmutableArray<ParameterSymbol> parameters = methodOrIndexer.GetParameters(); 
+
+            ArrayBuilder<IArgument> argumentsInEvaluationBuilder = ArrayBuilder<IArgument>.GetInstance(parameters.Length);
+
+            bool canSkipRewriting = CanSkipRewriting(ref arguments, methodOrIndexer, expanded, argsToParamsOpt, invokedAsExtensionMethod, out var isComReceiver); 
+
+            if (canSkipRewriting)
+            {
+                // In this case, the invocation is not in expanded form and there's no named argument provided.
+                // So we just return list of arguments as is.   
+                for (int i = 0; i < arguments.Length; i++)
+                {                                                 
+                    argumentsInEvaluationBuilder.Add(BoundCall.CreateArgumentOperation(ArgumentKind.Explicit, parameters[i], arguments[i]));
+                } 
+                return argumentsInEvaluationBuilder.ToImmutableAndFree(); ;
+            }                                                                                                                   
+
+            ArrayBuilder<ParameterSymbol> missingParametersBuilder = ArrayBuilder<ParameterSymbol>.GetInstance(parameters.Length);
+
+            BuildExplicitArgumentsInEvaluationOrder(syntax, 
+                methodOrIndexer, 
+                expanded, 
+                argsToParamsOpt, 
+                arguments, 
+                binder, 
+                argumentsInEvaluationBuilder, 
+                missingParametersBuilder);
+
+            AppendMissingOptionalArguments(syntax, methodOrIndexer, optionalParametersMethod, expanded, binder, missingParametersBuilder, argumentsInEvaluationBuilder);
+            missingParametersBuilder.Free();
+
+            return argumentsInEvaluationBuilder.ToImmutableAndFree();
+        }
+
+        // temporariesBuilder will be null when factory is null.
+        private static bool CanSkipRewriting(
+            ref ImmutableArray<BoundExpression> rewrittenArguments,
```

Comment:
probably because it was needed before I moved a chunk of code out of this method. Will remove.